collections/_posts/2024-12-02-SSL_comparison.md

```diff
@@ -0,0 +1,123 @@
+
+# Introduction
+
+In medical domain there is often a lack of annotated data for deep learning model training while there is a lot of unlabeled data in data health records. Semi-supervised and self-supervised learning are methods developed to take advantage of these unlabeled images to improve classification accuracy. The first one train classifiers jointly with two loss terms, while the second is a two-stages approach, the first to learn deep representation and the second to fine-tune the classifier. However, the two methods are rarely compared. The first question they try to answer is: **Which recent semi- or self-supervised methods are likely to be most effective?** \
+But those methods are very sensitive to hyperparameters, and benchmarks often don't consider this (either no hyperparameters tuning or tuning with a huge labeled validation set, bigger than the training set, which is not realistic). In this paper, they try to take that in account by answering the question: **Given limited available labeled data and limited compute, is hyperparameter tuning worthwhile?**
+
+# Methods
+
+## Definition of SSL
+
+A unified perspective for different learning paradigms is presented with two available datasets: $$L$$ which is a small labeled dataset and $$U$$ the unlabeled dataset with $$ \lvert U \rvert >> \lvert L \rvert $$. The learning process can be written as: 
+
+$$ v^*, w^* ← argmin_{v,w} \sum_{x,y \in L} \lambda^L l^L(y, g_w(f_v(x))) + \sum_{x \in U} \lambda^U l^U(x, f_v,g_w) $$ 
+
+where $$l^L$$ and $$l^U$$ are the loss functions linked to the labeled and unlabeled datasets and $$\lambda^L$$ and $$\lambda_{U}$$ the associated weights. In addition, $$f_v(\cdot)$$  denotes a neural network backbone with parameters $$v$$ that produces an embedding of the input images and $$g_w(\cdot)$$ denotes a final linear softmax classification layer with parameters $$w$$.
+
+| Method          | $$\lambda^L$$ | $$\lambda^U$$ |
+|-----------------|---------------|---------------|
+| Supervised      | 1             | 0             |
+| Semi-supervised | >0            | >0            |
+| Self-supervised | 0 then 1      | 1 then 0      |
+
+## Compared methods
+
+**Supervised learning**
+
+Only using the labeled $$(L)$$ dataset.
+
+* Sup: to denote a classifier trained with classical multiclass cross entropy loss.
+* MixUp: also multiclass cross entropy, but with mixup data augmentation (creating new training samples by mixing two samples from the original dataset).
+* SupCon: using a supervised contrastive learning loss (pull together samples belonging to the same class in the embedded space while pushing apart from samples of other classes).
+
+**Semi-supervised learning**
+
+They selected 6 methods of the recent years to cover a wide spectrum of strategies and cost, to take advantage of the unlabeled $$(U)$$ and labeled $$(L)$$ datasets jointly.
+
+* Pseudo-label: sets the label to unlabeled data to the class which has the maximum predicted probability.
```

Comment:
"sets the label to unlabeled data to the class..." → Suggested: "assigns labels to unlabeled data based on the class with the highest predicted probability."